core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java

```diff
@@ -2116,6 +2087,26 @@ public int getMaxSIRepairLimit(String dbName, String tableName) {
     return Math.abs(Integer.parseInt(thresholdValue));
   }
 
+  /**
+   * The below method returns the microseconds after which the trash folder will expire
+   */
+  public long getTrashFolderExpirationTime() {
+    String configuredValue = getProperty(CarbonCommonConstants.CARBON_TRASH_EXPIRATION_DAYS,
+            CarbonCommonConstants.CARBON_TRASH_EXPIRATION_DAYS_DEFAULT);
+    Integer result = 0;
+    try {
+      result = Integer.parseInt(configuredValue);
+      if (result < 0) {
+        LOGGER.warn("Value of carbon.trash.expiration.days is negative, taking default value");
+        result = Integer.parseInt(CARBON_TRASH_EXPIRATION_DAYS_DEFAULT);
+      }
+    } catch (NumberFormatException e) {
+      LOGGER.error("Error happened while parsing", e);
```

Comment:
```suggestion
      LOGGER.error("Invalid value configured for " + "CarbonCommonConstants.CARBON_TRASH_EXPIRATION_DAYS, considering the default value", e);
```

Please refactor it, in case of failure also it should take default value, now it will be considered as 0, and common message you can take to a variable to reduce the code and keep clean.

Reply:
done